java-security/src/main/java/com/sap/cloud/security/adapter/xs/XSUserInfoAdapter.java

```diff
@@ -34,40 +48,52 @@
 	static final String ZDN = "zdn";
 	static final String SYSTEM = "SYSTEM";
 	static final String HDB = "HDB";
+
 	private static final Logger LOGGER = LoggerFactory.getLogger(XSUserInfoAdapter.class);
-	private final AccessToken accessToken;
-	private OAuth2ServiceConfiguration configuration;
 
-	public XSUserInfoAdapter(Object accessToken) throws XSUserInfoException {
-		this(accessToken, null);
-	}
+	private final AccessToken accessToken;
+	private final OAuth2ServiceConfiguration configuration;
+	private final OAuth2TokenService oAuth2TokenService;
 
-	public XSUserInfoAdapter(Token accessToken) throws XSUserInfoException {
-		this(accessToken, Environments.getCurrent().getXsuaaConfiguration());
+	public XSUserInfoAdapter(Object accessToken) {
+		this(accessToken, Environments.getCurrent().getXsuaaConfiguration(), new DefaultOAuth2TokenService());
 	}
 
-	public XSUserInfoAdapter(AccessToken accessToken) throws XSUserInfoException {
-		this(accessToken, null);
+	public XSUserInfoAdapter(AccessToken accessToken) {
+		this(accessToken, Environments.getCurrent().getXsuaaConfiguration(), new DefaultOAuth2TokenService());
```

Comment:
this is incompatible: new DefaultOAuth2TokenService() requires apache rest client, but this is only "provided"... there might be other projects, that like to use Spring RestTemplate 

Question: is apache rest client provided as part of the SAP buildpack?

Reply:
The apache rest client is provided internally by the same class loader like `XSUserInfoAdapter`. It is not on the class path of the application however. I am not 100% sure if this is a problem and will check if that works in a sample/arquillian test. 

If someone uses the XSUserInfoAdapter standalone **without** buildpack, then they would need to bring their own apache rest client.

Reply:
For buildpack users that just works now.